firebase-firestore/src/main/java/com/google/firebase/firestore/core/Query.java

```diff
@@ -458,8 +464,13 @@ public int compare(Document doc1, Document doc2) {
     }
   }
 
-  /** @return A {@code Target} instance this query will be mapped to in backend and local store. */
-  public Target toTarget() {
+  /**
+   * This method is marked as synchronized to prevent the memoizedTarget from being
```

Comment:
Similarly to above comment, we can just state that this might change internal state.

Reply:
updated